components/dashboard/src/data/current-user/authenticated-user-query.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { useQuery } from "@tanstack/react-query";
+import { userClient } from "../../service/public-api";
+import { GetAuthenticatedUserRequest, User } from "@gitpod/public-api/lib/gitpod/v1/user_pb";
+
+export const useAuthenticatedUser = () => {
+    const query = useQuery<User>({
+        queryKey: getAuthenticatedUserQueryKey(),
```

Comment:
is it fine that we don't have any retry specify, there is some by default?

Reply:
We'll come to that later. Opted to keep UserContext for now, thus there is no change. `useAuthenticatedUser` isn't a complete drop-in replaced yet.